### PR TITLE
Handle mutable warnings from static analyzer in Core code

### PR DESCRIPTION
**Pull request by Dr15Jones**
#### PR description:

Went through the usage of mutable in most of the Core code. Either removed the need for the mutable or added the appropriate thread-safety attribute.

#### PR validation:

Code compiles.

### CODE REVIEW

**Comment by Dr15Jones** [comment]
please test

File: FWCore/Framework/src/ProductResolvers.h [hunk 1]
```diff
@@ -307,12 +308,14 @@ namespace edm {
...
-    mutable WaitingTaskList waitingTasks_;
+    CMS_THREAD_SAFE mutable WaitingTaskList waitingTasks_;
     mutable std::atomic<bool> prefetchRequested_;
     // for provenance
     ParentageID parentageID_;
     // for filter in a Path
-    mutable ProductStatus status_;
+    // The variable is only modified or read at times where the
+    //  framework has guaranteed synchronization between write and read
+    CMS_THREAD_SAFE mutable ProductStatus status_;
```
Comment (makortel) [inline FWCore/Framework/src/ProductResolvers.h]: Following our chat, this one likely should be `std::atomic<ProductStatus>` (as `DataManagingOrAliasProductResolver::theStatus_` is, but I'll take a closer look (=remind myself) how exactly it is being used (e.g. if there is a need for synchronization between threads).

**Comment by Dr15Jones** [comment]
+1

File: FWCore/Framework/src/ProductResolvers.h (hunk 1 above)
Comment (Dr15Jones) [reply FWCore/Framework/src/ProductResolvers.h]: We will address this issue in another pull request.

**Comment by fabiocos** [comment]
+1

**Pull request merged**